clients/iceberg-views/src/test/java/org/apache/iceberg/nessie/BaseIcebergTest.java

```diff
@@ -80,13 +80,14 @@ public BaseIcebergTest(String branch) {
 
   private void resetData() throws NessieConflictException, NessieNotFoundException {
     for (Reference r : api.getAllReferences().get().getReferences()) {
-      if (r instanceof Branch) {
+      // remove only 'branch' that is used by the test.
+      if (r instanceof Branch && r.getName().equals(branch)) {
```

Comment:
I'd prefer to delete all branches, that are not the default branch `main`. Otherwise tests that create more branches with the same name would start to fail.